Makefile

```
@@ -31,10 +31,12 @@ override FLAGS += $(if $(release),--release )$(if $(stats),--stats )$(if $(progr
 SPEC_WARNINGS_OFF := --exclude-warnings spec/std --exclude-warnings spec/compiler
 SPEC_FLAGS := $(if $(verbose),-v )$(if $(junit_output),--junit_output $(junit_output) )
 CRYSTAL_CONFIG_BUILD_COMMIT := $(shell git rev-parse --short HEAD 2> /dev/null)
+SOURCE_DATE_EPOCH := $(shell git show -s --format=%ct HEAD 2> /dev/null)
```

No, I mean, why is this changing the behavior where the default previously was the current date but now it's something from a commit
```suggestion
SOURCE_DATE_EPOCH := $(shell date +%s)
```

---

Because that's what SOURCE_DATE_EPOCH is supposed to be. It should represent the date of the source code, not the date of the build.

---

OK but this is 2 separate changes in one pull request, just makes it harder to agree

---

Why would you prefer it the other way around?

---

Hm, maybe.

---

But actually I don't think so. Without changing to source based date by default, even I would probably prefer #9057 over just externalizing `date +%s`.
With this change it is clear that determining the value should be handeld outside the compiler.

---

Can we try falling back to `stat -f "%c" Makefile` for tarball based builds?

---

For GNU stat it's `stat -c "%Y" Makefile`, so it's not portable. Neither is `date -r Makefile +%s`.

---

MacOS seems to error out on `stat -c "%Y" Makefile`, so we can just chain the approaches?

```
(git show -s --format=%ct HEAD  || stat -c "%Y" Makefile ||  stat -f "%m" Makefile) 2> /dev/null